monitoring/otel/config/otel-collector-config.yml

```diff
@@ -0,0 +1,183 @@
+              [
+                {
+                  type: latency,
+                  latency: { threshold_ms: 300000 }
+                },
+                # Only upload operations
+                {
+                  type: string_attribute,
+                  string_attribute: {
+                    key: http.operation,
+                    values: [ .*upload.* ],
+                    enabled_regex_matching: true,
+                  }
+                }
+              ]
+          }
+        },
+        # Sample traces for tenants with default mode
+        # Default mode is the mode where the trace.mode attribute is set to basic
+        # and only 0.2 of traces are sampled for each tenant
+        {
+          name: sampling-basic-tenants,
+          type: and,
+          and: {
+            and_sub_policy:
+              [
+                {
+                  # must have tenant.ref attribute
+                  name: has-tenant-ref,
+                  type: string_attribute,
+                  string_attribute:
+                    {
+                      key: tenant.ref,
+                      values: [ .* ],
+                      enabled_regex_matching: true
+                    },
+                },
+                {
+                  # trace.mode = basic
+                  name: trace-mode-default,
+                  type: string_attribute,
+                  string_attribute:
+                    {
+                      key: trace.mode,
+                      values: [ basic ],
+                    },
+                },
+                {
+                  name: success-status-codes,
+                  type: numeric_attribute,
+                  numeric_attribute: { key: http.status_code, min_value: 200, max_value: 399 }
+                },
+                {
+                  name: basic-sampling,
+                  type: probabilistic,
+                  probabilistic: {
+                    sampling_percentage: 5
+                  }
+                }
+              ]
+          }
+        },
+
+        # Sample traces for tenants with premium mode
+        # Premium mode sample 100% of traces for each tenant
+        {
+          name: sampling-premium-tenants,
+          type: and,
+          and: {
+            and_sub_policy:
+              [
+                {
+                  # must have tenant.ref attribute
+                  name: has-tenant-ref,
+                  type: string_attribute,
+                  string_attribute:
+                    {
+                      key: tenant.ref,
+                      values: [ .* ],
+                      enabled_regex_matching: true
+                    },
+                },
+                {
+                  # trace.mode = premium
+                  name: trace-mode-default,
+                  type: string_attribute,
+                  string_attribute:
+                    {
+                      key: trace.mode,
+                      values: [ premium ],
```

Comment:
premium is a weird term for this. enhanced?

Reply:
Sure, can change to `enhanced` or `always_trace`